js/src/forum/ForumApplication.js

```diff
@@ -94,6 +95,8 @@ export default class ForumApplication extends Application {
 
     this.pane = new Pane(document.getElementById('app'));
     this.composer = m.mount(document.getElementById('composer'), Composer.component());
+    this.cache.discussionList = new DiscussionListState();
+    this.cache.discussionList.refresh();
```

Comment:
You are calling .refresh() twice here - once in the constructor, and the second time here.

You should *not* be refreshing the discussion list in the mounting anyway. That's a pointless API call if they are not anywhere with the discussion list visible.

Reply:
Why pointless: wouldn't it just preload all the discussions? What would be a better place for it?

I suppose we could always take out the refresh method from the constructor if we want to preserve the state instance throughout the lifetime of app?

Reply:
Because that's an API call that will occur on every page load. If you are not in the IndexPage, and are not planning to go there, it's an extra call that shouldn't be occurring.

Also - it doesn't only affect the IndexPage. It'd be preloading the default discussion list, and if the current page is a tag discussion list, the response of that API call would be discarded (as it doesn't have the query params and all).

We shouldn't be making API calls when the resources requested aren't needed by the user.

Reply:
true, fixed.